### PR TITLE
Ports profiling improvements from TG

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

Ports https://github.com/tgstation/tgstation/pull/54894 and https://github.com/tgstation/tgstation/pull/59581 and https://github.com/tgstation/tgstation/pull/48861 and https://github.com/tgstation/tgstation/pull/54809

~~DNM until I test it.~~

## Changelog
:cl: actioninja, LemonInTheDark, and MrStonedOne
code: Performance is now logged to a CSV
code: SendMaps can now be profiled with a Debug verb.
code: Improved SSprofiler, and makes it only run every 5 minutes
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
@Crossedfall You'd probably be interested in using this data when you're no longer busy

**Comment by PowerfulBacon** [comment]
I didn't even know we profiled ever

File: code/controllers/subsystem/time_track.dm [hunk 1]
```diff
@@ -35,4 +104,59 @@ SUBSYSTEM_DEF(time_track)
...
+			MAPTICK_LAST_INTERNAL_TICK_USAGE,
+			length(SStimer.timer_id_dict),
+			SSair.cost_turfs,
+			SSair.cost_groups,
+			SSair.cost_highpressure,
+			SSair.cost_hotspots,
+			SSair.cost_superconductivity,
+			SSair.cost_pipenets,
+			SSair.cost_rebuilds,
+			length(SSair.active_turfs),
```
Comment (ike709) [inline code/controllers/subsystem/time_track.dm]:
```suggestion
```

File: code/controllers/subsystem/time_track.dm [hunk 2]
```diff
@@ -35,4 +104,59 @@ SUBSYSTEM_DEF(time_track)
...
+			length(SSair.hotspots),
+			length(SSair.networks),
+			length(SSair.high_pressure_delta),
+			length(SSair.active_super_conductivity)
```
Comment (ike709) [inline code/controllers/subsystem/time_track.dm]:
```suggestion
```

**Comment by ike709** [comment]
Someone review this

File: code/__DEFINES/subsystems.dm [hunk 3]
```diff
@@ -120,6 +120,7 @@
 #define INIT_ORDER_AI_CONTROLLERS 	55 //So the controller can get the ref
 #define INIT_ORDER_TICKER			55
 #define INIT_ORDER_MAPPING			50
+#define INIT_ORDER_TIMETRACK		47
```
Comment (PowerfulBacon) [inline code/__DEFINES/subsystems.dm:123]: Is this used anywhere?

Comment (ike709) [reply code/__DEFINES/subsystems.dm:123]: image

Bruh

**Comment by ike709** [comment]
Fine.

**Pull request merged**